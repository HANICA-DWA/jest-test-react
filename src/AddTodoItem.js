import React from 'react'

const AddTodoItem = (props) => {
  const onSubmit = (e) => {
    e.preventDefault()
    props.addTodo({name: e.target.todo.value})
    e.target.todo.value = ""
  }

  return <form onSubmit={onSubmit}>
    <input name="todo" />
  </form>
}

export default AddTodoItem 