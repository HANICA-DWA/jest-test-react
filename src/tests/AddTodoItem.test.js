import React from 'react'
import AddTodoItem from '../AddTodoItem'
import { renderer, shallow } from './testhelpers'

test('AddTodoItem adds fires addTodo onSubmit', () => {
  const todos = []
  const addTodo = (todo) => {
    todos.push(todo)
  }
  const component = shallow(
    <AddTodoItem addTodo={addTodo} />
  )
  component.simulate('submit', {
    preventDefault: () => { },
    target: {todo: { value: 'Hello'}}
  })
  expect(todos.length).toBe(1)
})

test('AddTodoItem is rendered', () => {
  const comp = renderer.create(<AddTodoItem />)
  expect(comp.toJSON()).toMatchSnapshot();
})