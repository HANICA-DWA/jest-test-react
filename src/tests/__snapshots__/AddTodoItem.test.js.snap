// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodoItem is rendered 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="todo"
  />
</form>
`;
